### PR TITLE
Add validation to VertexBuffer::build()

### PR DESCRIPTION
**Pull request by prideout**
Clients should be prevented from allocating unused buffer slots because
uploading to an unused slot triggers undefined behavior in the backend.

Motivated by #2042

### CODE REVIEW

**Comment by prideout** [comment]
@pixelflinger are you ok with this?

File: filament/src/VertexBuffer.cpp [hunk 1]
```diff
@@ -123,6 +123,19 @@ VertexBuffer* VertexBuffer::Builder::build(Engine& engine) {
         return nullptr;
     }
 
+    auto const& declaredAttributes = mImpl->mDeclaredAttributes;
+    auto const& attributes = mImpl->mAttributes;
+    utils::bitset32 mAttributedBuffers;
```
Comment (pixelflinger) [inline filament/src/VertexBuffer.cpp]: It's kind of weird to have a local variable with the `m` prefix -- was it the intention?

File: filament/src/VertexBuffer.cpp [hunk 2]
```diff
@@ -123,6 +123,19 @@ VertexBuffer* VertexBuffer::Builder::build(Engine& engine) {
         return nullptr;
     }
 
+    auto const& declaredAttributes = mImpl->mDeclaredAttributes;
```
Comment (pixelflinger) [inline filament/src/VertexBuffer.cpp]: I think this whole section deserves a comment explaining what it checks and why.

File: filament/src/VertexBuffer.cpp (hunk 1 above)
Comment (prideout) [reply filament/src/VertexBuffer.cpp]: Woops, fixed.

File: filament/src/VertexBuffer.cpp (hunk 2 above)
Comment (prideout) [reply filament/src/VertexBuffer.cpp]: Done.

**Comment by prideout** [comment]
@pixelflinger I made the requested changes.

**Pull request merged**